### PR TITLE
WPF - Add support for Shift+AltGr

### PR DESCRIPTION
**Pull request by ptrsky**
The handling of AltGr characters in CefBrowsterHostWrapper did not cover capital diacritic letters. This commit adds handling for the Shift + AltGr + modifier combination.

**Fixes: #3626**

**Summary:**
   - I have added support for Shift+AltGr character input in CefSharp WPF

**Changes:** 
   - I have modified the current handling of AltGr characters in the CefBrowsterHostWrapper class by extending support to Shift+Ctrl+Alt combination
      
**How Has This Been Tested?**  
   - ran CefSharp.WPF.Example
   - navigate to www.google.com
   - enter capital polish letters (Shift + AltGr + EOASLZXCN) 
   - enter small polish letters (AltGr + eoaslzxcn), 
   - enter special characters (Shift + 1-9)

**Screenshots (if appropriate):**
obraz

**Types of changes**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Updated documentation

**Checklist:**

- [x] Tested the code(if applicable)
- [x] Commented my code
- [ ] Changed the documentation(if applicable)
- [ ] New files have a license disclaimer
- [x] The formatting is consistent with the project (project supports .editorconfig)

### CODE REVIEW

File: CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp [hunk 1]
```diff
@@ -391,10 +391,19 @@ void CefBrowserHostWrapper::SendKeyEvent(int message, int wParam, int lParam)
             // 2 Either CTRL key is pressed.
```
Comment (amaitland) [inline CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp:391]: Ideally this comment is updated to include the new Shift behaviour.

File: CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp [hunk 2]
```diff
@@ -391,10 +391,19 @@ void CefBrowserHostWrapper::SendKeyEvent(int message, int wParam, int lParam)
...
+            if (((scan_res >> 8) & 0xFF) == ctrlAlt) // ctrl-alt pressed
+            {
+                keyEvent.modifiers &= ~(EVENTFLAG_CONTROL_DOWN | EVENTFLAG_ALT_DOWN);
+                keyEvent.modifiers |= EVENTFLAG_ALTGR_DOWN;
+            }
+            else if (((scan_res >> 8) & 0xFF) == shiftCtrlAlt) // shift-ctrl-alt pressed
             {
                 keyEvent.modifiers &= ~(EVENTFLAG_CONTROL_DOWN | EVENTFLAG_ALT_DOWN);
                 keyEvent.modifiers |= EVENTFLAG_ALTGR_DOWN;
+                keyEvent.modifiers |= EVENTFLAG_SHIFT_DOWN;
```
Comment (amaitland) [inline CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp:406]: Can you confirm that modifiers needs shift explicitly set, I would have been expecting that `GetCefKeyboardModifiers` would have already set the flag.

**Review by amaitland — CHANGES_REQUESTED** [review]
Thanks for the `PR` 👍 

Comments inline

File: CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp (hunk 1 above)
Comment (ptrsky) [reply CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp:392]: Done

File: CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp (hunk 2 above)
Comment (ptrsky) [reply CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp]: You're right - the shift modifier is already set and there's no need to set it again. I got rid of the "else if" statement and simply expanded the original check to include the combination with Shift.

Comment (amaitland) [reply CefSharp.Core.Runtime/Internals/CefBrowserHostWrapper.cpp]: Great 👍

**Comment by amaitland** [comment]
We can probably simplify this as suggested in https://bitbucket.org/chromiumembedded/cef/pull-requests/377#Ltests/cefclient/browser/osr_window_win.ccT808

**Pull request merged**